### PR TITLE
test: Validation operations on Closing Stock balance

### PR DESCRIPTION
**Pull request by Prateek8848**
**Title**
Validation operations on Closing Stock balance
**Description**
Test Summary
This commit adds a comprehensive test case, test_set_status_TC_SCK_292, designed to validate the lifecycle operations of the Closing Stock Balance doctype in the Frappe framework. The test ensures that the document transitions correctly through various states such as Draft, Submitted, and Cancelled, and that regeneration logic works as expected. This contributes to enhancing test coverage of the Stock module and improving reliability.

**Scenarios Covered**
Input Validations:
Checks for the existence of necessary master data (Company, Item, Warehouse) and creates them if missing to ensure the test environment is ready.

Lifecycle State Changes:
Tests the correct behavior when submitting a Closing Stock Balance document, canceling it, and regenerating the closing stock balance.

**Edge Cases:**
Handles repeated creation and submission to verify consistent behavior and no unexpected failures on re-submission.

**API / Method Handling:**
Validates the document’s state changes triggered by calling the submit(), cancel(), and regenerate_closing_balance() methods, asserting the document status after each action.

**Technology**
Frappe Test Framework — Utilizes Frappe’s built-in testing tools for document creation, submission, cancellation, and assertion.

**Linked Feature/Bug**
Not linked to any particular feature or bug.

### CODE REVIEW

File: erpnext/stock/doctype/closing_stock_balance/test_closing_stock_balance.py [hunk 1]
```diff
@@ -1,9 +1,74 @@
...
+				"item_code": item,
+				"include_uom": "Box",
+				"warehouse": warehouse,
+				"item_group": "All Item Groups",
+				"warehouse_type": "Transit",
+			}
+		).insert(ignore_permissions=True)
+		closing_balance.submit()
+
+		closing_balance = frappe.get_doc(
```
Comment (aasif-patel) [inline erpnext/stock/doctype/closing_stock_balance/test_closing_stock_balance.py]: var closing_balance are defined two times?

File: erpnext/stock/doctype/closing_stock_balance/test_closing_stock_balance.py [hunk 2]
```diff
@@ -1,9 +1,74 @@
...
+	def tearDown(self):
+		frappe.db.rollback()
+
+	# codecov
+	def test_set_status_TC_SCK_292(self):
+		item_code = "Test Item"
+		company = "_Test Indian Registered Company"
+		warehouse = "'Stores - _TC'"
+		# Ensure prerequisites exist
+		if not frappe.db.exists("Company", company):
```
Comment (aasif-patel) [inline erpnext/stock/doctype/closing_stock_balance/test_closing_stock_balance.py:21]: please check the method creating records are handling the duplicate entry or not, if not then only call db exists

**Review by aasif-patel — CHANGES_REQUESTED** [review]

**Comment by aasif-patel** [comment]
@Prateek8848 add proper title

**Comment by Prateek8848** [comment]
> @Prateek8848 add proper title

@aasif-patel Added proper title and removed checking item exist and added no pragma to ignore lines ,and removed duplicate creation of closing stcok entry,please check

**Review by aasif-patel — APPROVED** [review]

**Comment by swathibollineni2** [comment]
@Prateek8848 Test case passed
